src/main/webapp/app/exercises/shared/submission-policy/submission-policy-update.component.ts

```diff
@@ -108,7 +108,7 @@ export class SubmissionPolicyUpdateComponent implements OnInit {
                 Validators.required,
             ]),
             exceedingPenalty: new FormControl({ value: this.programmingExercise.submissionPolicy?.exceedingPenalty, disabled: !this.editable }, [
-                Validators.min(0),
+                Validators.min(0.000000000000000000000001),
```

Comment:
This value seems to be a bit random to me. Is there a reason why it changed to explicitly this value?

Reply:
The client-side states that values equal to 0 are fine. But trying to create a programming exercise with a value of 0, an exception after clicking 'Generate' is thrown ("error.submissionPenaltyPolicyIllegalPenalty") and the exercise will not be created. But I agree that this is some kind of a bodgy solution. Added a regex for validating this instead.